### PR TITLE
Workflow temp1 (#380)

### PR DESCRIPTION
**Pull request by matthewchungcodefresh**
* Adding kaniko-add-template

* revert changes

* Update

* Update

* Fix indentation

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Rename template to jira

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* change rbac name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* repository update

* fix spelling error in command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* remove file

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Update output variables

* Update output

* Update Dockerfile

* Update

* Update workflowTemplate.yaml

* fixing vulnerabilities

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* using alpine image instead

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Fix template name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* update templates

* Update workflowTemplate.yaml

* changes from classic to CSDP

* Update workflowTemplate.yaml

* Jira complete

* Update workflowTemplate.yaml

* Pull request for jira integration template (#329)

* Adding kaniko-add-template

* revert changes

* Update

* Update

* Fix indentation

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Rename template to jira

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* change rbac name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* repository update

* fix spelling error in command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* remove file

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Update output variables

* Update output

* Update Dockerfile

* Update

* Update workflowTemplate.yaml

* fixing vulnerabilities

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* using alpine image instead

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Fix template name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* update templates

* Update workflowTemplate.yaml

* changes from classic to CSDP

* Update workflowTemplate.yaml

* Jira complete

* Update workflowTemplate.yaml

Co-authored-by: Laurent <laurent.rochette@codefresh.io>

* Initial commit

* Update sonar-scanner.md

* Version 1.0.0 complete

* Added/Modified workflows/jira/sonar.sensor.yaml, workflows/jira/cdp-csdp-matthew-hub.ingress.yaml, workflows/jira/sonar.event-source.yaml, workflows/jira/sonar-test.workflow-template.yaml

* update

* add command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* add command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Added/Modified workflows/sonar.sensor.yaml, workflows/cdp-csdp-matthew-hub.ingress.yaml, workflows/sonar.event-source.yaml, workflows/sonar-test.workflow-template.yaml

* remvoe test template

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* remvoe test template

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* rename template

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* test on Tue Jun 21 17:16:08 MDT 2022

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* test on Tue Jun 21 17:17:55 MDT 2022

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Added/Modified workflows/testing.sensor.yaml, workflows/cdp-csdp-matthew-hub.ingress.yaml, workflows/testing.event-source.yaml, workflows/argo-hub.ci-simple.0.0.1.workflow-template.yaml

* Added/Modified workflows/testing-2.sensor.yaml, workflows/cdp-csdp-matthew-hub.ingress.yaml, workflows/testing-2.event-source.yaml

* Sonar Update

* svg changes

* remove test files

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Fix stuff

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Update Dockerfile

* Jira fixed readme links

* Update workflowTemplate.yaml

* datree update

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Sonar update

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* V0.0.1 commit

* Delete workflows/aqua directory

* Delete workflows/registry directory

* delete files

* Sonar template for CSDP (#367)

* Adding kaniko-add-template

* revert changes

* Update

* Update

* Fix indentation

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Rename template to jira

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* change rbac name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* repository update

* fix spelling error in command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* remove file

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Update output variables

* Update output

* Update Dockerfile

* Update

* Update workflowTemplate.yaml

* fixing vulnerabilities

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* using alpine image instead

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Fix template name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* update templates

* Update workflowTemplate.yaml

* changes from classic to CSDP

* Update workflowTemplate.yaml

* Jira complete

* Update workflowTemplate.yaml

* Pull request for jira integration template (#329)

* Adding kaniko-add-template

* revert changes

* Update

* Update

* Fix indentation

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Rename template to jira

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* change rbac name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* repository update

* fix spelling error in command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* remove file

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Update output variables

* Update output

* Update Dockerfile

* Update

* Update workflowTemplate.yaml

* fixing vulnerabilities

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* using alpine image instead

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Fix template name

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* update templates

* Update workflowTemplate.yaml

* changes from classic to CSDP

* Update workflowTemplate.yaml

* Jira complete

* Update workflowTemplate.yaml

Co-authored-by: Laurent <laurent.rochette@codefresh.io>

* Initial commit

* Update sonar-scanner.md

* Version 1.0.0 complete

* Added/Modified workflows/jira/sonar.sensor.yaml, workflows/jira/cdp-csdp-matthew-hub.ingress.yaml, workflows/jira/sonar.event-source.yaml, workflows/jira/sonar-test.workflow-template.yaml

* update

* add command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* add command

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Added/Modified workflows/sonar.sensor.yaml, workflows/cdp-csdp-matthew-hub.ingress.yaml, workflows/sonar.event-source.yaml, workflows/sonar-test.workflow-template.yaml

* remvoe test template

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* remvoe test template

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* rename template

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* test on Tue Jun 21 17:16:08 MDT 2022

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* test on Tue Jun 21 17:17:55 MDT 2022

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Added/Modified workflows/testing.sensor.yaml, workflows/cdp-csdp-matthew-hub.ingress.yaml, workflows/testing.event-source.yaml, workflows/argo-hub.ci-simple.0.0.1.workflow-template.yaml

* Added/Modified workflows/testing-2.sensor.yaml, workflows/cdp-csdp-matthew-hub.ingress.yaml, workflows/testing-2.event-source.yaml

* Sonar Update

* svg changes

* remove test files

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Fix stuff

Signed-off-by: Laurent <laurent.rochette@codefresh.io>

* Update Dockerfile

* Jira fixed readme links

* Update workflowTemplate.yaml

* datree update

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Sonar update

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* V0.0.1 commit

* Delete workflows/aqua directory

* Delete workflows/registry directory

* Delete argo-hub.ci-simple.0.0.1.workflow-template.yaml

* Delete cdp-csdp-matthew-hub.ingress.yaml

* Delete testing-2.event-source.yaml

* Delete testing-2.sensor.yaml

* Delete testing.event-source.yaml

* Delete testing.sensor.yaml

* Delete workflows/datree directory

* Delete workflows/jira directory

Co-authored-by: Laurent <laurent.rochette@codefresh.io>
Co-authored-by: matthewchungcodefresh <>

* Fixed issues

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Update sonar-scanner.md

* Fix Bugs

* Delete workflows/datree directory

* Delete argo-hub.ci-simple.0.0.1.workflow-template.yaml

* Delete cdp-csdp-matthew-hub.ingress.yaml

* Delete testing-2.event-source.yaml

* Delete testing-2.sensor.yaml

* Delete testing.event-source.yaml

* Delete testing.sensor.yaml

* Delete workflows/jira directory

* Update workflowTemplate.yaml

* Update workflowTemplate.yaml

* Workflow template added

* Update icon.svg

* Delete workflows/sonar directory

Co-authored-by: Laurent <laurent.rochette@codefresh.io>
Co-authored-by: matthewchungcodefresh <>

### CODE REVIEW

**Review by lrochette — CHANGES_REQUESTED** [review]
the first paragraph to the CHANGELOG needs to be changed. It's a little confusing

**Review by lrochette — COMMENTED** [review]
There are way too many parameters.
Can you imagine what will need to be done for something trying to create a CI using the pipeline wizard?
Try to reduce the number of parameters to 10 or by making assumptions.
For example 
* For Jira authentication, simply use 1 secret and make assumption on the key used (token, username, .....) and hardcode them on the call where needed
* same for Sonar
* Hardcode the ISSUE project and customfields (so there is an example). I don't expect those to be picked in a real scenario from the webhook payload but either be hardcoded to calculated during the run
* the image SHA should be grabbed from the build (it's not something you know at the start of the workflow)
* ....

Make sense?

File: workflows/jira-scan-workflow/versions/0.0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+## Templates
+
+1. [jira-sonar-template](https://github.com/codefresh-io/argo-hub/blob/main/workflows/jira-scan-template/versions/0.0.1/docs/jira-sonar-template.md) 
+
+## Security
+
+Minimal required permissions
+
+[Full rbac permissions list](./rbac.yaml)
```
Comment (eti-codefresh) [inline workflows/jira-scan-workflow/versions/0.0.1/README.md:15]: please give full path

File: workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+    argo-hub/version: '0.0.1'
+    argo-hub/description: 'Workflow template for a full deployment using jira, sonar and a canary rollout'
+    argo-hub/categories: 'argo'
+    argo-hub/license: 'MIT'
+    argo-hub/owner_name: 'Matthew Chung'
+    argo-hub/owner_email: 'matthew.chung@codefresh.io'
+    argo-hub/owner_avatar: 'https://github.com/matthewchungcodefresh.png'
+    argo-hub/owner_url: 'https://github.com/matthewchungcodefresh'
+    argo-hub/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/workflows/jira-scan-workflow/assets/icon.svg"
+    argo-hub/icon-background: "#f4f4f4"
```
Comment (eti-codefresh) [inline workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:15]: argo-hub/icon_background: "#f4f4f4"

File: workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+      inputs:
+        parameters:
+          - name: IMAGE
+      container:
+        image: '{{ inputs.parameters.IMAGE }}'
+        command: [sh, -c]
+        args: ["ls"]
+
+    # canary rollout template 10%
+    - name: rollout-10
```
Comment (eti-codefresh) [inline workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:498]: need to add 
      metadata:
        annotations:

File: workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+        image: codefresh/cf-deploy-kubernetes:master
+        workingDir: /tmp/repo
+        commands:
+          - /cf-deploy-kubernetes ./canary-manual-approval/service.yaml 
+          - /cf-deploy-kubernetes ./canary-manual-approval/service-preview.yaml
+          - /cf-deploy-kubernetes ./canary-manual-approval/service-all.yaml  
+          - /cf-deploy-kubernetes ./canary-manual-approval/rollout.yaml   
+
+    # canary rollout template 33%
+    - name: rollout-33
```
Comment (eti-codefresh) [inline workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:515]: need to add
metadata:
annotations:

File: workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+            path: /tmp/repo
+        parameters:
+          - name: IMAGE
+      container:
+        image: kostiscodefresh/kubectl-argo-rollouts:latest
+        workingDir: /tmp/repo
+        commands:
+          - /app/kubectl-argo-rollouts-linux-amd64 promote golang-sample-app-deployment -n canary --context "mydemoAkscluster@BizSpark Plus"
+    
+    # canary rollout template 100%
```
Comment (eti-codefresh) [inline workflows/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:528]: need to add
metadata:
annotations:

**Review by eti-codefresh — CHANGES_REQUESTED** [review]

**Comment by lrochette** [comment]
@eti-codefresh was this template not supposed to go in the examples folder?
This is not a plugin!

**Pull request closed without merging**